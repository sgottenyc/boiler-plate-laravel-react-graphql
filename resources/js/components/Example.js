import React from 'react';
import PropTypes from 'prop-types';
/*
 *  JSX Below will create this component
 *  <Example title="" initialCount="0"/>
 */

export default class Example extends React.Component {
    constructor(props) {
      super(props);            
      this.onClick = this.onClick.bind(this);  //bind the function to this object
      this.state = { timesClicked: props.initialCount }  //set initial count
    }
    onClick() {
        this.setState( { timesClicked: this.state.timesClicked + 1 });
        alert('i clicked');
    }    
    render() {
        const { name, description } = this.props;       // destructure props object to local variable for use in jsx render function
        const { timesClicked } = this.state;            // destructure state object to local variable for use in jsx render function
        return (
            <div className="container">
                <div className="row justify-content-center">
                    <div className="col-md-8">
                        <div className="card">
                            <div className="card-header">{name}</div>
                            <div className="card-body" onClick={this.onClick}>                                                     
                                Number of times clicked {timesClicked}.
                            </div>                            
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

/*** USE PROP TYPES TO DEFINE DEFAULTS */
Example.propTypes = {
    name: PropTypes.string.isRequired,
    initialCount: PropTypes.number
};

/** SET DEFAULT STARTING VALUES */
Example.defaultProps = {
    initialCount: 0
};