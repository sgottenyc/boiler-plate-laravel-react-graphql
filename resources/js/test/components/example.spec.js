import React from 'react';
import Example from '../../components/Example';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

test('Render Example', () => {
  const component = renderer.create(
    <Example name="I am example" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

/* Example for testing with enzyme
test('Example changes after click', () => {
  // Render a checkbox with label in the document
  const example = shallow(<Example name="I am example" />);

  expect(checkbox.text()).toEqual('Off');

  example.find('input').simulate('click');

  expect(checkbox.text()).toEqual('On');
});
*/