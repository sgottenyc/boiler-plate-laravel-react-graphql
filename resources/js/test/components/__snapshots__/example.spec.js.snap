// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Example 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-md-8"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          I am example
        </div>
        <div
          className="card-body"
          onClick={[Function]}
        >
          Number of times clicked 
          0
          .
        </div>
      </div>
    </div>
  </div>
</div>
`;
